packages/backend/server/src/plugins/copilot/context/resolver.ts

```diff
@@ -664,29 +702,55 @@ export class CopilotContextResolver {
     @Args('content') content: string,
     @Args('limit', { type: () => SafeIntResolver, nullable: true })
     limit?: number,
+    @Args('scopedThreshold', { type: () => Float, nullable: true })
+    scopedThreshold?: number,
     @Args('threshold', { type: () => Float, nullable: true })
     threshold?: number
   ): Promise<ContextMatchedFileChunk[]> {
     if (!this.context.canEmbedding) {
       return [];
     }
 
-    const session = await this.context.get(context.id);
-
     try {
-      return await session.matchFileChunks(
+      if (!context.id) {
+        return await this.context.matchWorkspaceFiles(
+          context.workspaceId,
+          content,
+          limit,
+          this.getSignal(ctx.req),
+          threshold
+        );
```

Comment:
Should not return early here? It also needs to match workspace files when context id exists.

Reply:
workspace query logic has already been included in `matchFiles`